WordPress/UITestsFoundation/Screens/Editor/AztecEditorScreen.swift

```diff
@@ -11,63 +13,44 @@ public class AztecEditorScreen: BaseScreen {
     }
 
     let mode: Mode
-    var textView: XCUIElement
-
-    private var richTextField = "aztec-rich-text-view"
-    private var htmlTextField = "aztec-html-text-view"
-
-    let editorCloseButton = XCUIApplication().navigationBars["Azctec Editor Navigation Bar"].buttons["Close"]
-    let publishButton = XCUIApplication().buttons["Publish"]
-    let publishNowButton = XCUIApplication().buttons["Publish Now"]
-    let moreButton = XCUIApplication().buttons["more_post_options"]
-    let uploadProgressBar = XCUIApplication().progressIndicators["Progress"]
-
-    let titleView = XCUIApplication().textViews["Title"]
-    let contentPlaceholder = XCUIApplication().staticTexts["aztec-content-placeholder"]
-
-    let mediaButton = XCUIApplication().buttons["format_toolbar_insert_media"]
-    let insertMediaButton = XCUIApplication().buttons["insert_media_button"]
-    let headerButton = XCUIApplication().buttons["format_toolbar_select_paragraph_style"]
-    let boldButton = XCUIApplication().buttons["format_toolbar_toggle_bold"]
-    let italicButton = XCUIApplication().buttons["format_toolbar_toggle_italic"]
-    let underlineButton = XCUIApplication().buttons["format_toolbar_toggle_underline"]
-    let strikethroughButton = XCUIApplication().buttons["format_toolbar_toggle_strikethrough"]
-    let blockquoteButton = XCUIApplication().buttons["format_toolbar_toggle_blockquote"]
-    let listButton = XCUIApplication().buttons["format_toolbar_toggle_list_unordered"]
-    let linkButton = XCUIApplication().buttons["format_toolbar_insert_link"]
-    let horizontalrulerButton = XCUIApplication().buttons["format_toolbar_insert_horizontal_ruler"]
-    let sourcecodeButton = XCUIApplication().buttons["format_toolbar_toggle_html_view"]
-    let moreToolbarButton = XCUIApplication().buttons["format_toolbar_insert_more"]
-
-    let unorderedListOption = XCUIApplication().buttons["Unordered List"]
-    let orderedListOption = XCUIApplication().buttons["Ordered List"]
-
-    // Action sheets
-    let actionSheet = XCUIApplication().sheets.element(boundBy: 0)
-    let postSettingsButton = XCUIApplication().sheets.buttons["Post Settings"]
-    let keepEditingButton = XCUIApplication().sheets.buttons["Keep Editing"]
-    let postHasChangesSheet = XCUIApplication().sheets["post-has-changes-alert"]
```

Comment:
I recently did the opposite of this on the WooCommerce iOS repo where I moved them all to `let`s 😅 I thought having it grouped together would be easier to manage and see and also in case of reusability (though I have not work on the tests long enough to really know how much of the identifiers are being reused at this point)

I don't have strong opinions on either style, so I'm flexible but I am looking to see if we can get more consistency for the UI tests across apps so that it's easier to move between apps when writing UI tests.

Reply:
> I don't have strong opinions on either style, so I'm flexible but I am looking to see if we can get more consistency for the UI tests across apps so that it's easier to move between apps when writing UI tests.

This is gold. I settled for that approach for the reason stated above and also to keep the code more compact.

Regarding `let` vs runtime evaluation: My mental representation of how XCTest behaves is that it runs the query to fetch the element when it's defined. I developed this when bumping into some test failures that I resolved by moving to that style. I haven't built a dedicated project to verify this assumption, but I ought to.

If my assumption is incorrect, then my biggest argument in favor of not using `let`s would no longer be valid. 🤔 

Your point about having them "grouped together would be easier to manage and see and also in case of reusability" is totally valid. I personally prefer short, local code, and might have been a bit idealistic in imagining all our `ScreenObject` subclass will remain so short that it would be trivial to notice duplication. There's a lot of value in optimizing code to make it easier to touch in the future, so maybe I've been over-optimizing on the wrong dimension.

I guess the next step would be to verify that assumption on how the `XCUIElement`s are evaluated. With an answer to that, we can then decide what's best for us going forward.